compiler/AST/primitive.cpp

```diff
@@ -715,3 +716,12 @@ VarSymbol* newMemDesc(const char* str) {
   memDescsVec.add(s);
   return memDescVar;
 }
+
+VarSymbol* newMemDesc(Symbol* sym) {
```

Comment:
The argument could be a Type* too and that would be just as good as a TypeSymbol* in my opinion.